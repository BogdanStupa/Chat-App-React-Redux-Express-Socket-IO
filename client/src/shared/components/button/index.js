import React from "react";
import classNames from "classnames";


function ButtonComponent(props){
    const {
        text,
        primary,
        secondaryButton,
        link,
        isFetching,
        disabled,
        outline,
        marginTop,
        height
    } = props;

    const styleButton = classNames({
        btn: true,
        "btn--default": true && !link,
        "btn--link": link,
        "btn--disabled": disabled,
        "btn--primary": !disabled && primary
    });

    const onClick = () => {

    }
    return (
        <button
            className={styleButton}
            disabled={isFetching}
            onClick={onClick}
            style={{
                marginTop,
                height
            }}
        >
            {
                isFetching && !disabled 
                ? "loading"
                : text
            }
        </button>
        
    );
}

export default ButtonComponent;
