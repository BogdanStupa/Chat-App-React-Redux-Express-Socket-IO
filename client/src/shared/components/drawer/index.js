import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import classNames from "classnames";
import { createSelector } from "reselect";

import {
    ButtonComponent,
    IconComponent
} from "shared/components";
import { initDrawer, closeDrawer } from "redux/actions/drawer";

const selectIsOpen = createSelector(
    [ 
        (state,drawerName) => state.drawer[drawerName] || { [drawerName]: { isOpen: false }}
    ],
    (drawerNames) => drawerNames.isOpen || false
);


const DrawerComponent = props =>{
    const {
        children,
        title,
        icon,
        drawerName
    } = props;

    const dispatch = useDispatch();
    
    useEffect(() => {
        dispatch(initDrawer(drawerName));
    },[]);

    const isOpen = useSelector(state => selectIsOpen(state, drawerName));

    const drawerStyles = classNames({
        drawer: true,
        open: isOpen 
    });

    const onClick = () => dispatch(closeDrawer(drawerName));

    return (
        <div className={drawerStyles}>
            {
                isOpen && 
                <div className="drawer-container">
                    <header className="drawer-header">
                        <div className="drawer-header-content">
                            <ButtonComponent
                                link
                                onClick={onClick}
                            >
                                
                                <IconComponent
                                    icon={icon}
                                    width={30}
                                    height={28}
                                />
                                
                            </ButtonComponent>  
                            <h3>{ title }</h3>
                        </div>        
                    </header>
                    {
                        children
                    }
                </div> 
            }
        </div>
    )
}



export default DrawerComponent;