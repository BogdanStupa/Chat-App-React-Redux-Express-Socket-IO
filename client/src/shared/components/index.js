import ButtonComponent from "./button";
import InputComponent from "./input";
import LabelComponent from "./label";

export {
    ButtonComponent,
    InputComponent,
    LabelComponent
};
