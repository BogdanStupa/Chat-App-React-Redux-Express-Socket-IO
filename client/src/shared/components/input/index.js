import React from "react";

function InputComponent(props){
    const {
        value,
        onChange,
        type
    } = props;

    return (
        <div className="imput-wrapper">
            <input
                type={type}
                onChange={onChange}
                value={value}
            />
        </div>
        
    );
}

export default InputComponent;