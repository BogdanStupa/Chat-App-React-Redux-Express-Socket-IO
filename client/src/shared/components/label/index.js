import React from "react";
import ClassNames from "classnames";


const LabelComponent = React.memo( props => {
    const {
        text,
        children
    } = props;

    return (
    <label className="label">
        {text}
        {children}
    </label>
    );
});

export default LabelComponent;