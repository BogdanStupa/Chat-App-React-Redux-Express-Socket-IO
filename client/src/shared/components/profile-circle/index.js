import React from "react";


const ProfileCircle = props => {
    const {
        profile: {
            label,
            backgroundColor,
            width,
            height,
            fontSize
        } 
    } =  props;
    const profileText = label ? label.slice(0,2) : null; //test this on undefined

    return (
        <div 
            className="user-info-profile" 
            style={{
                backgroundColor: backgroundColor,
                width:width,
                height:height,
                minWidth: width,
                minHeight: height,
                fontSize: fontSize
            }}>
                {profileText}
        </div>
    );
}

export default ProfileCircle;