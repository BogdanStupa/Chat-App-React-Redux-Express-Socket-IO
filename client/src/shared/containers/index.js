import AppContainer from "./app";
import FormContainer from "./form";
import AuthContainer from "./auth";

export {
    AppContainer,
    FormContainer,
    AuthContainer
};