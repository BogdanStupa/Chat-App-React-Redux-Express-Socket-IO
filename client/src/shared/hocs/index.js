import withAuthRedirect from "./auth-redirect";

export {
    withAuthRedirect
};