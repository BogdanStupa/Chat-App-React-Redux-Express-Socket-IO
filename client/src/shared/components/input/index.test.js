import React from "react";
import { InputFormComponent } from "shared/components";
import { shallow } from  "enzyme";
import { componentCheckOnRender } from "tests-utils";

describe("<InputFormCamponent/>:", () =>{
    componentCheckOnRender("InputFormComponent", <InputFormComponent />);

    describe("is working correctly",() => {
        const onChange = jest.fn();
        const errors = "test error";
        const modelI = {
            type:"text",
            isFetching: false,
            onChange
        };
        const modelII = {
            type:"text",
            isFetching: false,
            onChange,
            errors
        }
        const modelIII = {
            type:"text",
            isFetching: true,
            onChange,
        };
        const getWrapper = model => {
            return shallow(<InputFormComponent { ...model }/>);
        };

        it("has input component", () => {
            const wrapper = getWrapper(modelI);
            expect(wrapper.find("input").length).toBe(1);
        });
        
        it("has an error component, where errors is defined", () => {
            const wrapper = getWrapper(modelII);
            expect(wrapper.find(".error-info").length).toBe(1);
        });

        it("has no error component, where no error", () => {
            const wrapper = getWrapper(modelI);
            expect(wrapper.find(".error-info").length).toBe(0);
        });

        it("correctly working onChange",() => {
            const wrapper = getWrapper(modelI);
            wrapper.find("input").simulate("change");
            expect(onChange).toHaveBeenCalled();
        });
    });
});
