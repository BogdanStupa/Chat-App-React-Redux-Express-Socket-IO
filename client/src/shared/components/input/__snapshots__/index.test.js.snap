// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputFormCamponent/>: InputFormComponentit should be renderd matches the snapshots 1`] = `"<div class=\\"input-wrapper\\" style=\\"font-size:1.2rem\\"><input class=\\"\\"/></div>"`;
